bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareBridgeHandler.java

```diff
@@ -0,0 +1,213 @@
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareHousehold;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcarePet;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SurePetcareBridgeHandler} is responsible for handling the bridge things created to use the Sure Petcare
+ * API. This way, the user credentials may be entered only once.
+ *
+ * It also spawns 2 background polling threads to update the more static data (topology) and the pet locations at
+ * different time intervals.
+ *
+ * @author Rene Scherer - Initial Contribution
+ */
+@NonNullByDefault
+public class SurePetcareBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SurePetcareBridgeHandler.class);
+
+    private final SurePetcareAPIHelper petcareAPI;
+    private @Nullable ScheduledFuture<?> topologyPollingJob;
+    private @Nullable ScheduledFuture<?> petStatusPollingJob;
+
+    public SurePetcareBridgeHandler(Bridge bridge, SurePetcareAPIHelper petcareAPI) {
+        super(bridge);
+        this.petcareAPI = petcareAPI;
+    }
+
+    @Override
+    // @SuppressWarnings("null")
+    public void initialize() {
+        logger.debug("Initializing Sure Petcare bridge handler.");
+        SurePetcareBridgeConfiguration config = getConfigAs(SurePetcareBridgeConfiguration.class);
+
+        if (config.username != null && config.password != null) {
+            updateStatus(ThingStatus.UNKNOWN);
+            try {
+                logger.debug("Login to SurePetcare API with username: {}", config.username);
+                petcareAPI.login(config.username, config.password);
+                logger.debug("Login successful, updating topology cache");
+                petcareAPI.updateTopologyCache();
+                logger.debug("Cache update successful, setting bridge status to ONLINE");
+                updateStatus(ThingStatus.ONLINE);
+                updateThings();
+            } catch (AuthenticationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/offline.conf-error-authentication");
+                return;
+            }
+        } else {
+            logger.warn("Setting thing '{}' to OFFLINE: Parameter 'password' and 'username' must be configured.",
+                    getThing().getUID());
```

Comment:
Since you are already setting the thing offline with a similar message you don't need to log it since thing status changes are automatically logged.

Reply:
done